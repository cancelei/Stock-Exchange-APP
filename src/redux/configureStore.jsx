import { configureStore } from '@reduxjs/toolkit';
import stocksReducers from './stocks/stockSlice';

export default configureStore({
  reducer: {
    stocks: stocksReducers,
  },
});
