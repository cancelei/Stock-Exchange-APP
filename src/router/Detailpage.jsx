import { useParams } from 'react-router-dom';
import { Container, Table } from 'react-bootstrap';
import Navbar from '../components/NavElement/NavElement';
import './DetailPage.css'; // Importing the CSS

export default function Detailpage() {
  const {
    symbol, name, price, change, dayLow, dayHigh, marketCap, volume, exchange,
  } = useParams();

  return (
    <main className="detail-main">
      <Navbar />
      <Container className="detail-container">
        <h2 className="text-center mb-4">
          Stock Detail for
          {' '}
          {name}
          (
          {symbol}
          )
        </h2>
        <Table striped bordered hover className="detail-table">
          <tbody>
            <tr>
              <td>Price</td>
              <td>
                {price}
              </td>
            </tr>
            <tr>
              <td>Daily change</td>
              <td>
                {change}
              </td>
            </tr>
            <tr>
              <td>Min value today</td>
              <td>
                {dayLow}
              </td>
            </tr>
            <tr>
              <td>Max value today</td>
              <td>
                {dayHigh}
              </td>
            </tr>
            <tr>
              <td>Market Cap</td>
              <td>
                {marketCap}
              </td>
            </tr>
            <tr>
              <td>Volume</td>
              <td>
                {volume}
              </td>
            </tr>
            <tr>
              <td>Exchange</td>
              <td>
                {exchange}
              </td>
            </tr>
          </tbody>
        </Table>
      </Container>
    </main>
  );
}
