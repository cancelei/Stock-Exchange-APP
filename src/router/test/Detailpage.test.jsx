import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../../redux/configureStore';
import '@testing-library/jest-dom/extend-expect';
import Detailpage from '../Detailpage';

describe('Tests the details page', () => {
  const component = (
    <Provider store={configureStore}>
      <BrowserRouter>
        <Detailpage />
      </BrowserRouter>
    </Provider>
  );

  it('Match the snapshot', () => {
    const tree = render(component).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
