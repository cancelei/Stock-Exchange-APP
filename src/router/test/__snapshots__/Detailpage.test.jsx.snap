// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the details page Match the snapshot 1`] = `undefined`;
